src/handle_action.cpp

```diff
@@ -741,19 +741,20 @@ static void smash()
                           player_character.get_wielded_item()->attack_time( player_character ) *
                           0.8;
     bool mech_smash = false;
-    int smashskill;
+    int smashskill = player_character.get_arm_str();
     ///\EFFECT_STR increases smashing capability
     if( player_character.is_mounted() ) {
         auto *mon = player_character.mounted_creature.get();
-        smashskill = player_character.get_arm_str() + mon->mech_str_addition() + mon->type->melee_dice *
-                     mon->type->melee_sides;
+        smashskill += mon->mech_str_addition() + mon->type->melee_dice * mon->type->melee_sides;
         mech_smash = true;
-    } else {
-        smashskill = player_character.get_arm_str();
-        if( player_character.get_wielded_item() ) {
-            smashskill += player_character.get_wielded_item()->damage_melee( damage_bash );
-        }
-    }
+    } else if( player_character.get_wielded_item() ) {
+        smashskill += player_character.calculate_by_enchantment(
```

Comment:
Why only if the player is weilding a weapon?

Reply:
If you mean "why do I only factor in MELEE_DAMAGE enchantment if the player is wielding a weapon", then you're right, this is a mistake. I should be linearly adding smashskill by the bash rating of the weapon they're currently holding and then perform the enchantment calculation afterwards.

I think there are two ways I could do it from there: the first is to call 

```
smashskill = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::MELEE_DAMAGE,
                  true );
smashskill = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::ITEM_DAMAGE_BASH,
                  true );
```

or, I could do something like

```
int melee_damage_enchantment =  player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::MELEE_DAMAGE,
                  true );
int item_damage_bash_enchantment = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::ITEM_DAMAGE_BASH,
                  true );
smashskill += melee_damage_enchantment + item_damage_bash_enchantment
```

I'm leaning towards the second one but would like to hear your thoughts.

Reply:
I think #1 is easier to read 😄